packages/core-mobile/app/new/features/stake/components/StakesScreen.tsx

```diff
@@ -0,0 +1,166 @@
+  const { theme } = useTheme()
+  const isDevMode = useSelector(selectIsDeveloperMode)
+  const { networkToken: pChainNetworkToken } =
+    NetworkService.getAvalancheNetworkP(isDevMode)
+
+  const completeCardBackground = theme.isDark
+    ? require('../../../assets/icons/complete-card-bg-dark.png')
+    : require('../../../assets/icons/complete-card-bg-light.png')
+
+  const claimableInAvax = useGetClaimableBalance()
+
+  const data: StakeCardType[] = useMemo(() => {
+    const result = [StaticCard.Add]
+    if (claimableInAvax?.gt(0.05)) {
+      result.push(StaticCard.Claim)
+    }
+    return [...result, ...stakes]
+  }, [stakes, claimableInAvax])
+
+  const renderItem = useCallback(
+    ({
+      item,
+      index
+    }: ListRenderItemInfo<StakeCardType>): JSX.Element | null => {
+      let content = null
+      if (item === StaticCard.Add) {
+        content = <AddCard width={CARD_WIDTH} onPress={onAddStake} />
+      } else if (item === StaticCard.Claim) {
+        content = (
+          <ClaimCard
+            onPress={onClaim}
+            title={`${claimableInAvax?.toDisplay({
+              fixedDp: 2
+            })} AVAX reward unlocked`}
+            width={CARD_WIDTH}
+            backgroundImageSource={completeCardBackground}
+          />
+        )
+      } else {
+        const now = new Date()
+
+        if (isCompleted(item, now)) {
+          const rewardAmountInAvaxDisplay = formattedRewardAmountInAvax(item)
+
+          content = (
+            <CompletedCard
+              onPress={() => onPressStake(item.txHash)}
+              title={`${rewardAmountInAvaxDisplay} AVAX reward claimed`}
+              width={CARD_WIDTH}
+            />
+          )
+        } else if (isOnGoing(item, now)) {
+          const remainingTime = getReadableDateDuration(
+            new UTCDate(secondsToMilliseconds(item.endTimestamp || 0))
+          )
+          const estimatedRewardInAvaxDisplay = formattedEstimatedRewardInAvax(
+            item,
+            pChainNetworkToken
+          )
+
+          content = (
+            <ProgressCard
+              title={`${estimatedRewardInAvaxDisplay} AVAX reward unlocked in\n${remainingTime}`}
+              progress={getActiveStakeProgress(item, now)}
+              width={CARD_WIDTH}
+              motion={motion}
+              onPress={() => onPressStake(item.txHash)}
+            />
+          )
+        }
+      }
+
+      if (content) {
+        return (
+          <Animated.View
+            entering={getListItemEnteringAnimation(index)}
+            layout={SPRING_LINEAR_TRANSITION}>
+            {content}
+          </Animated.View>
+        )
+      }
+
+      return null
+    },
+    [
+      claimableInAvax,
+      completeCardBackground,
+      motion,
+      onAddStake,
+      onClaim,
+      onPressStake,
+      pChainNetworkToken
+    ]
+  )
+
+  return (
+    <CollapsibleTabs.FlatList
+      contentContainerStyle={styles.container}
+      data={data}
+      numColumns={2}
```

Comment:
I think we should use flashlist here since number of stakes can grow pretty large (we don't have a limit and fetch all stakes right now)